accepted/0000-remove-dependency-on-fulcio.md

```diff
@@ -0,0 +1,69 @@
+# RFC for linking packages to their source and build without using Sigstore Fulcio
+
+## Summary
+
+As it is showed in a recent paper that Sigstore Fulcio is an unnecessary trusted third party for linking npm packages to their source and build, we suggest the npm team consider the new approach presented in that paper which can be found at:
+
+https://eprint.iacr.org/2023/003
+
+The new approach not only eliminates the unnecessary dependency on an external black-box CA but also accelerates the signing process.
+
+## Motivation
+
+Last year an RFC for linking npm packages to their source and build using Sigstore Fulcio was proposed and accepted:
+
+https://github.com/npm/rfcs/pull/626
+
+However, the paper mentioned above shows the same result can be achieved without using Sigstore Fulcio at all. Since relying on an unnecessary black-box CA poses a serious threat to the npm community, we draft this RFC.
+
+## Detailed Explanation
+
+The key observation made by the paper mentioned above is the following:
+
+- Instead of setting the Audience claim of a requested OIDC token to be `sigstore` (as proposed by the previous RFC), a CI/CD workflow can in fact set the Audience claim to be something like `sigstore?pk=PubKey`, where `PubKey` is (the Base64 encoding of) the public key used by the CI/CD workflow for signing the software artifact.
+
+In other words, such an OIDC token can be understood as a witness signed by the CI/CD service, with the Audience claim encoding the workflow's intention to use the public key to represent it. Thus, instead of exchanging an OIDC token (and a public key) for a Fulcio certificate, we can directly use the OIDC token in place of the certificate (which is, after all, transcribed from the OIDC token).
```

Comment:
To be clear: this entails publishing the OIDC token as proof, right?

I understand the witness proof provided by doing so, but it's worth noting that this is potentially dangerous to do: OIDC tokens are just JWTs under the hood, and audience confusion/logic errors are a well-known class of bugs with JWTs. Even if the token is only short-lived, this could potentially result in lateral movement/unintended access to systems that accept OIDC tokens but fail to adequately filter based on audience.

Reply:
To use an example:

1. Alice uses GitHub Actions to interact with both Amazon Web Services and Sigstore. Both use OIDC to perform authentication; Amazon Web Services uses the `sub` claim and Sigstore uses the `aud` claim.
2. Alice performs a Sigstore signing operation, and publishes her OIDC token per the amended scheme here.
3. Mallory takes the OIDC token made public by Alice and uses it to impersonate Alice on AWS, gaining access to whatever resources the token's `sub` claims allow.

Reply:
(TL;DR this requires every OIDC RP to handle the `aud` claim, when many will just check `sub` or some other claim(s) relevant to their purposes.)

Reply:
Mallory cannot impersonate Alice on AWS if Alice configures AWS correctly by requiring a check of the audience claim, as showed in the official GitHub Actions document [here](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services). But I understand your point is not every Alice is knowledgeable enough to do this right. Thank you for your insight.

However, the Alice in the scope of our discussion is a repository owner/collaborator, and I tend to believe she should be knowledgeable enough. After all, if she cannot do this right, there are more things she could do wrong with her resources. And the bottom line is the security of her resources should not be based on hiding the OIDC tokens.